### PR TITLE
allow user to name usb key when adding

### PR DESCRIPTION
**Pull request by hobbitronics**
- user adds a usb security key name before touch.vue

Needs translations!

If continue is clicked without input
image

if a duplicate name is entered
image

After inserting the key the input appears and autofocuses
image

### CODE REVIEW

**Comment by Baggerone** [comment]
When I enter a name for my new (and only) security key, I get this in the javascript console.
```vue.runtime.esm.js?c320:4605 [Vue warn]: Error in v-on handler: "TypeError: Cannot read properties of undefined (reading 'find')"

found in

---> <VBtn>
       <ButtonBar> at src/global/ButtonBar.vue
         <VStepperContent>
           <VStepper>
             <ProfileWizard> at src/profile/ProfileWizard.vue
               <Insert> at src/2sv/key/Insert.vue
                 <VMain>
                   <VApp>
                     <App> at src/App.vue
                       <Root>```

**Comment by hobbitronics** [comment]
> When I enter a name for my new (and only) security key, I get this in the javascript console.
> 
> ```
> 
> found in
> 
> ---> <VBtn>
>        <ButtonBar> at src/global/ButtonBar.vue
>          <VStepperContent>
>            <VStepper>
>              <ProfileWizard> at src/profile/ProfileWizard.vue
>                <Insert> at src/2sv/key/Insert.vue
>                  <VMain>
>                    <VApp>
>                      <App> at src/App.vue
>                        <Root>```
> ```

Should be fixed now.

**Comment by Baggerone** [comment]
Please don't merge this until we get the base changes rolled out to production.
Devon said we could do that now.

**Comment by Baggerone** [comment]
This works for me locally when I use the correct feature docker tags in the docker-compose file.
I'll see about getting my branches PR'd as well.

File: src/2sv/key/Insert.vue [hunk 1]
```diff
@@ -21,26 +27,75 @@
...
+      if(event.key == 'Enter') {
+        this.onContinue()
+      }
+    },
+    onContinue: function() {
+      if(!this.input.trim()) {
+        this.snackBarMessage = this.$vuetify.lang.t('$vuetify.2sv.key.insert.label')
+        this.snackbarIsOpen = true
+        return
+      } else if (mfa.keys.data?.find(key => key.label == this.input.trim())) {
```
Comment (forevermatt) [inline src/2sv/key/Insert.vue:90]: While this looks like it works, would you mind extracting the logic in this if-condition to a separate function, for readability? Something like this...
```js
} else if (isDuplicateLabel(mfa.keys, attemptedKeyLabel)) {
```
... though that assumes extracting/assigning `this.input.trim()` to a new `attemptedKeyLabel` value near the top of onContinue.

Feel free to adjust the specifics however you want.

**Review by forevermatt — APPROVED** [review]
Approved, but with one request (not critical)

File: src/2sv/key/Insert.vue (hunk 1 above)
Comment (hobbitronics) [reply src/2sv/key/Insert.vue]: no worries. I had to call `this.isDuplicateLabel` which makes Vue start to feel a little old compared to Svelte.

**Pull request merged**